// @pulumi/aws/bedrock/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProvisionedModelThroughput = exports.getCustomModelsOutput = exports.getCustomModels = exports.getCustomModelOutput = exports.getCustomModel = exports.CustomModel = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
exports.CustomModel = null;
utilities.lazyLoad(exports, ["CustomModel"], () => require("./customModel"));
exports.getCustomModel = null;
exports.getCustomModelOutput = null;
utilities.lazyLoad(exports, ["getCustomModel", "getCustomModelOutput"], () => require("./getCustomModel"));
exports.getCustomModels = null;
exports.getCustomModelsOutput = null;
utilities.lazyLoad(exports, ["getCustomModels", "getCustomModelsOutput"], () => require("./getCustomModels"));
exports.ProvisionedModelThroughput = null;
utilities.lazyLoad(exports, ["ProvisionedModelThroughput"], () => require("./provisionedModelThroughput"));
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "aws:bedrock/customModel:CustomModel":
                return new exports.CustomModel(name, undefined, { urn });
            case "aws:bedrock/provisionedModelThroughput:ProvisionedModelThroughput":
                return new exports.ProvisionedModelThroughput(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("aws", "bedrock/customModel", _module);
pulumi.runtime.registerResourceModule("aws", "bedrock/provisionedModelThroughput", _module);
